runtime/major_gc.c

```diff
@@ -614,7 +614,12 @@ Caml_inline uintnat rotate1(uintnat x)
   return (x << ((sizeof x)*8 - 1)) | (x >> 1);
 }
 
-Caml_noinline static intnat do_some_marking(intnat work)
+Caml_noinline static intnat do_some_marking
+#ifndef CAML_INSTR
+  (intnat work)
+#else
+  (intnat work, int* slice_fields, int* slice_pointers)
```

Comment:
There are already a lot of ifdefs around. Would something be lost by removing the conditional (and silence unused variable warning using `(void)var;` instead)?

Reply:
I don't think that would really help. The CAML_INSTR code is very ifdeffy anyway.

Reply:
Removing this ifdef would lead to removing ifdefs in several other places, this is why I was thinking it makes things simpler.